### DISCUSSION TITLE
Detect Newly Connected FTDI Devices

### DISCUSSION THREAD

**Discussion by sethwellsNZ** [eblot/pyftdi#357] [Q&A] [answered]
Hi there,

I'm having a little trouble with a specific use-case for this library.
I'm using a PCB with two FTDI devices on it, and externally the chips can be turned off and on.
I need to be able to detect just one chip when it is on, and then without re-executing the script, detect both chips after the second has been turned on.

Currently, I use Ftdi.find_all() to locate the chips. It works fine to detect the first chip, but then after powering on the second chip calling Ftdi.find_all() still only finds the first chip. I have to restart the script to get the function to detect the second chip too.

Anyone have any ideas? I've tried reloading and deleting the Ftdi module using importlib, to no success.

Thanks!

**Comment by sifive-eblot** [comment, marked as answer]
Have you tried to [flush the device cache](https://eblot.github.io/pyftdi/api/usbtools.html#pyftdi.usbtools.UsbTools.flush_cache)?

**Reply by sethwellsNZ** [reply]
That did the trick, thanks very much
